x/wasm-storage/keeper/msg_server.go

```diff
@@ -0,0 +1,85 @@
+package keeper
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/CosmWasm/wasmd/x/wasm/ioutils"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sedaprotocol/seda-chain/x/wasm-storage/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (k msgServer) StoreDataRequestWasm(goCtx context.Context, msg *types.MsgStoreDataRequestWasm) (*types.MsgStoreDataRequestWasmResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	unzipped := unzipWasm(msg.Wasm)
+	wasm := types.NewWasm(unzipped, msg.WasmType)
+	if k.Keeper.HasDataRequestWasm(ctx, wasm) {
```

Comment:
I think the goal is to be able to overwrite a hash since we are permissionless yes @mariocao @FranklinWaller?

Reply:
The hash is based on the bytes so overwriting is not a thing right

Reply:
@FranklinWaller Oh the `Wasm` struct has the field `WasmType`, which could be overwritten.

Reply:
We could just get rid of `WasmType` if it's unnecessary to store this info on chain.

Reply:
Yea, that's the only thing I was wondering about. I should have been more specific. My bad!

Eventually, if/when we add metadata too, that could be an issue... maybe rather than storing metadata and that stuff on the chain, it could emit an event or something for the indexer to grab with that info? i.e. we still take it as args, but we don't store it just give it to indexer in some way.

Reply:
Late to the party but I think the idea is to have the metadata be indexer only. So as long as you have the private key used to upload the bytes you can edit the metadata through the explorer.